lib/matplotlib/axes/_axes.py

```diff
@@ -3929,6 +3929,9 @@ def line_props_with_rcdefaults(subkey, explicit, zdelta=0):
             d = {k.split('.')[-1]: v for k, v in rcParams.items()
                  if k.startswith(f'boxplot.{subkey}')}
             d['zorder'] = zorder + zdelta
+            # override the marker only if it is set by rcParams['lines.marker']
+            # leave it unchanged otherwise (gh-15730)
+            d['marker'] = d.get('marker', '')
```

Comment:
You can also use `d.setdefault('marker', '')` which will only set the the value if the key does not yet exist.

Reply:
Thanks! handled this in a cleaner way based on the feedback given by the other developers (checking the existence of the key is no longer required because this is handled by explicitly setting an argument `use_marker`).